### ISSUE TITLE
[feature request] delta upload of locally modified files

### ISSUE THREAD

**Issue by l3iggs** [odeke-em/drive#61]
Let's say I have a 1000MB file locally and I push it up to my Google drive. Let's say the 'drive push' command takes 100 seconds to complete (I guess this is roughly the upload time). Now let's say I locally modify 100MB in that 1000MB file and run 'drive push' again. I expect this operation take 10 seconds to complete. My testing seems to show that this second push takes 100 seconds to complete.

Is my expectation unreasonable?
Can a delta upload feature be added easily?

**Comment by odeke-em** [comment]
That's going to take a while until I incorporate change tracking [which I already started working on https://github.com/odeke-em/drive/tree/change-tracking]. This is because Google Drive doesn't directly expose the ability to check the integrity of chunks but there is away around this by tracking changes. Please be patient, or if you have any suggestions feel free to provide them.

**Comment by l3iggs** [comment]
After doing a bit of reading, it looks like Google Drive doesn't support they type of delta sync I'm asking about here. Pity.

It seems like this is a feature request for Google rather than for this project at this point. I'm going to close it.

My gut tells me that effort to implement this feature in drive without proper API support from Google will be too much of a hack and is likely wasted effort :-(

**Comment by odeke-em** [comment]
Yes in deed it is a very tricky feature since as previously mentioned Google Drive doesn't support it, yet.

**Comment by kcwu** [comment]
FYI, http://librsync.sourcefrog.net/doc/rdiff.html

**Comment by odeke-em** [comment]
That is pretty cool, you deserve a cookie @kcwu! Surely something to take a look at.

**Comment by kcwu** [comment]
We cannot use it directly since we cannot control server side. I guess some ideas still could be reused.

**Comment by odeke-em** [comment]
Oh yes the ideas from this are useful. Since we are already talking about ideas: my ideas for solving this would involve constructing merkle trees or tiger hash trees and then comparing these from past revisions and figuring out what chunks changed. However I have a way simpler idea but it will require Google Drive support for which I have to go through discussions and proposals that might not be accepted but we shall see.

**Comment by greyltc** [comment]
Delta sync would be a really great feature. @odeke-em has there been any progress on this?

**Comment by odeke-em** [comment]
@greyltc nope, no progress unfortunately has been made on my part here.
